x/merkledb/db.go

```diff
@@ -479,9 +479,14 @@ func (db *merkleDB) PrefetchPath(key []byte) error {
 func (db *merkleDB) prefetchPath(view *trieView, keyBytes []byte) error {
 	return view.visitPathToKey(ToKey(keyBytes), func(n *node) error {
 		if !n.hasValue() {
+			if _, ok := db.intermediateNodeDB.nodeCache.Get(n.key); ok {
```

Comment:
We should add a comment that this is done to improve concurrent prefetching performance.